ionic2/app/pages/poke-detail/poke-detail.page.ts

```diff
@@ -27,4 +27,18 @@ export class PokeDetailPage {
     this.pokemon = navParams.get('pokemon');
   }
 
+  getRankInWords() {
+    let reverseRank = 152 - this.pokemon.rarityRank;
+
+    if (reverseRank == 1) {
+      return "1st";
+    } else if (reverseRank == 2) {
+      return "2nd";
+    } else if (reverseRank == 3) {
+      return "3rd";
+    } else {
+      return reverseRank + "th";
```

Comment:
That's not gonna work for
21 -> 21st
22 -> 22nd
23 -> 23rd
... and so on. I think this is what you need http://stackoverflow.com/a/13627586